src/chttpd/src/chttpd_changes.erl

```diff
@@ -604,31 +605,49 @@ keep_sending_changes(Args, Acc0, FirstRound) ->
                 true ->
                     case wait_updated(Timeout, TimeoutFun, UserAcc3) of
                         {updated, UserAcc4} ->
-                            UserCtx = fabric2_db:get_user_ctx(Db),
-                            DbOptions1 = [{user_ctx, UserCtx} | DbOptions],
-                            case fabric2_db:open(fabric2_db:name(Db), DbOptions1) of
-                                {ok, Db2} ->
-                                    ?MODULE:keep_sending_changes(
-                                        Args#changes_args{limit = NewLimit},
-                                        ChangesAcc#changes_acc{
-                                            db = Db2,
-                                            user_acc = UserAcc4,
-                                            seq = EndSeq,
-                                            prepend = Prepend2,
-                                            timeout = Timeout,
-                                            timeout_fun = TimeoutFun
-                                        },
-                                        false
-                                    );
-                                _Else ->
-                                    end_sending_changes(Callback, UserAcc3, EndSeq)
+                            AccumulatedTime = timer:now_diff(os:timestamp(), T0) div 1000,
```

Comment:
To avoid the `div 1000` could use the more modern (OTP 18+) time functions here:

```
1> erlang:system_time(millisecond).
1632492974842
2> T0 = erlang:system_time(millisecond), timer:sleep(1000), erlang:system_time(millisecond) - T0.
1000
```

Reply:
I tried to match the original diff as closely as possible but am happy to upgrade to these functions if we don't need to support anything older?

Reply:
Ah ok, it's not a big deal whichever looks better to you.

We don't support anything lower than OTP 21 on `main` and 20 on `3.x` so all the new time functions should be there already.

Reply:
os:timestamp() is used elsewhere in the same module so I think I'd rather leave this as-is, rather than change them all or have a combination of methods for the same thing.